Code/src/main/java/nl/utwente/group10/haskell/type/VarT.java

```diff
@@ -146,6 +146,9 @@ public final String toString() {
      * @return The set of type classes that are in both types.
      */
     public static Set<TypeClass> intersect(VarT a, VarT b) {
+        if (!a.hasConstraints()) return b.constraints;
+        if (!b.hasConstraints()) return a.constraints;
+        
```

Comment:
Heb er bijgezet dat het een tijdelijke workaround is, maar heb het er wel ingelaten omdat zonder deze fix de error display stuk gaat.